crates/noirc_evaluator/src/ssa_refactor/acir_gen/mod.rs

```diff
@@ -337,6 +325,42 @@ impl Context {
         }
     }
 
+    /// Returns the a vector of `AcirVar`s constrained to be result of the function call.
+    ///
+    /// The function being called is required to be intrinsic.
+    fn convert_ssa_call(
+        &mut self,
+        func: ValueId,
+        arguments: &[ValueId],
+        dfg: &DataFlowGraph,
+    ) -> Vec<AcirVar> {
+        let intrinsic = Self::id_to_intrinsic(func, dfg);
+        let black_box = match intrinsic {
+            Intrinsic::BlackBox(black_box) => black_box,
+            _ => todo!("expected a black box function"),
+        };
+        let mut inputs = Vec::new();
+        for value_id in arguments {
+            if Self::value_is_array_address(*value_id, dfg) {
+                let (array_id, index) = self
+                    .ssa_value_to_array_address
+                    .get(value_id)
+                    .expect("ICE: Call argument of undeclared array");
+                assert_eq!(index, &0, "ICE: Call arguments only accept arrays in their entirety");
+                let elements = self
+                    .acir_context
+                    .array_load_all(*array_id)
+                    .expect("add Result types to all methods so errors bubble up");
+                inputs.extend(elements);
+            } else {
+                inputs.push(self.convert_ssa_value(*value_id, dfg));
+            }
```

Comment:
I can see what you're saying about this check being a bit odd. Can it be reconciled if we pass `inputs` to `convert_ssa_value` such that the function can push multiple inputs if needed? E.g. `self.convert_ssa_value(*value_id, dfg, &mut inputs);`. Then that function could handle array addresses internally.

Reply:
Makes sense. Currently call instructions are the only place where we need to gather the contents of an array. Until that changes, I figure it's better not to force that interface change on all the other places that use `convert_ssa_value`.